### PR TITLE
Wip 351 streamlit test

### PR DESCRIPTION
**Pull request by PaulineMauryL**

### CODE REVIEW

**Comment by PaulineMauryL** [comment]
## Coverage report


 <details><summary>Click to see where and how coverage changed</summary><table><thead>
  <tr><th>File</th><th>Statements</th><th>Missing</th><th>Coverage</th><th>Coverage<br>(new stmts)</th><th>Lines missing</th></tr>
</thead>
<tbody><tr>
<td colspan="6">&nbsp;&nbsp;<b>server/lomas_server/admin_database</b></td><tr>
<td>&nbsp;&nbsp;<a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-3fca177177b39806c7ad155fa923f0bc494bd12882b88c19aca839deb4aee49c">constants.py</a></td>
<td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-3fca177177b39806c7ad155fa923f0bc494bd12882b88c19aca839deb4aee49c"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-3fca177177b39806c7ad155fa923f0bc494bd12882b88c19aca839deb4aee49c"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-3fca177177b39806c7ad155fa923f0bc494bd12882b88c19aca839deb4aee49c"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-3fca177177b39806c7ad155fa923f0bc494bd12882b88c19aca839deb4aee49c"></a></td><td></td><tr>
<td colspan="6">&nbsp;&nbsp;<b>server/lomas_server/administration/tests</b></td><tr>
<td>&nbsp;&nbsp;<a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-94c81326dcf36ee6582ced72bddc1aec0ee9e2cdc454540d3e8beebfef256ef3">test_streamlit_app.py</a></td>
<td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-94c81326dcf36ee6582ced72bddc1aec0ee9e2cdc454540d3e8beebfef256ef3"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-94c81326dcf36ee6582ced72bddc1aec0ee9e2cdc454540d3e8beebfef256ef3"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-94c81326dcf36ee6582ced72bddc1aec0ee9e2cdc454540d3e8beebfef256ef3"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-94c81326dcf36ee6582ced72bddc1aec0ee9e2cdc454540d3e8beebfef256ef3"></a></td><td><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-94c81326dcf36ee6582ced72bddc1aec0ee9e2cdc454540d3e8beebfef256ef3R12-R30">12-30</a>, <a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-94c81326dcf36ee6582ced72bddc1aec0ee9e2cdc454540d3e8beebfef256ef3R36-R60">36-60</a>, <a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-94c81326dcf36ee6582ced72bddc1aec0ee9e2cdc454540d3e8beebfef256ef3R66-R89">66-89</a></td><tr>
<td colspan="6">&nbsp;&nbsp;<b>server/lomas_server/models</b></td><tr>
<td>&nbsp;&nbsp;<a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-c12a4fdeafdc71621d7bdfcf31e4c5e720bb56aa597b102e9ecbe139979e8197">config.py</a></td>
<td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-c12a4fdeafdc71621d7bdfcf31e4c5e720bb56aa597b102e9ecbe139979e8197"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-c12a4fdeafdc71621d7bdfcf31e4c5e720bb56aa597b102e9ecbe139979e8197"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-c12a4fdeafdc71621d7bdfcf31e4c5e720bb56aa597b102e9ecbe139979e8197"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-c12a4fdeafdc71621d7bdfcf31e4c5e720bb56aa597b102e9ecbe139979e8197"></a></td><td></td><tr>
<td>&nbsp;&nbsp;<a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-34bdfee75b6160a429d2078bf6c5b90f15865be9b446435ce040ab611e04cdc8">constants.py</a></td>
<td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-34bdfee75b6160a429d2078bf6c5b90f15865be9b446435ce040ab611e04cdc8"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-34bdfee75b6160a429d2078bf6c5b90f15865be9b446435ce040ab611e04cdc8"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-34bdfee75b6160a429d2078bf6c5b90f15865be9b446435ce040ab611e04cdc8"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-34bdfee75b6160a429d2078bf6c5b90f15865be9b446435ce040ab611e04cdc8"></a></td><td></td><tr>
<td colspan="6">&nbsp;&nbsp;<b>server/lomas_server/tests</b></td><tr>
<td>&nbsp;&nbsp;<a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-7f7d18385b324497d5ebcf51c3e0c0172158683cd7f916a75a261815c51f617c">test_api.py</a></td>
<td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-7f7d18385b324497d5ebcf51c3e0c0172158683cd7f916a75a261815c51f617c"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-7f7d18385b324497d5ebcf51c3e0c0172158683cd7f916a75a261815c51f617c"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-7f7d18385b324497d5ebcf51c3e0c0172158683cd7f916a75a261815c51f617c"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-7f7d18385b324497d5ebcf51c3e0c0172158683cd7f916a75a261815c51f617c"></a></td><td></td><tr>
<td>&nbsp;&nbsp;<a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-104b12d96cc2aef41d54eba074122e74b72eb88673b73d285069b195421c9530">test_dummy_generation.py</a></td>
<td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-104b12d96cc2aef41d54eba074122e74b72eb88673b73d285069b195421c9530"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-104b12d96cc2aef41d54eba074122e74b72eb88673b73d285069b195421c9530"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-104b12d96cc2aef41d54eba074122e74b72eb88673b73d285069b195421c9530"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-104b12d96cc2aef41d54eba074122e74b72eb88673b73d285069b195421c9530"></a></td><td></td><tr>
<td>&nbsp;&nbsp;<a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-8860acb398146823e4b75c9bac2b9fc191e2f220076e2a3c57b9e20ffc2634bc">test_mongodb_admin.py</a></td>
<td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-8860acb398146823e4b75c9bac2b9fc191e2f220076e2a3c57b9e20ffc2634bc"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-8860acb398146823e4b75c9bac2b9fc191e2f220076e2a3c57b9e20ffc2634bc"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-8860acb398146823e4b75c9bac2b9fc191e2f220076e2a3c57b9e20ffc2634bc"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-8860acb398146823e4b75c9bac2b9fc191e2f220076e2a3c57b9e20ffc2634bc"></a></td><td></td><tr>
<td>&nbsp;&nbsp;<a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-7e12850d27b3e53e690a3a59c2dc3bf18468e648a98f2554b2781f882dfffc86">test_mongodb_admin_cli.py</a></td>
<td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-7e12850d27b3e53e690a3a59c2dc3bf18468e648a98f2554b2781f882dfffc86"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-7e12850d27b3e53e690a3a59c2dc3bf18468e648a98f2554b2781f882dfffc86"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-7e12850d27b3e53e690a3a59c2dc3bf18468e648a98f2554b2781f882dfffc86"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-7e12850d27b3e53e690a3a59c2dc3bf18468e648a98f2554b2781f882dfffc86"></a></td><td></td></tbody>
<tfoot>
<tr>
<td><b>Project Total</b></td>
<td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-4b0bf2efa3367c0072ac2bf1e234e703dc46b47aaa4fe9d3b01737b1a15752b1"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-4b0bf2efa3367c0072ac2bf1e234e703dc46b47aaa4fe9d3b01737b1a15752b1"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-4b0bf2efa3367c0072ac2bf1e234e703dc46b47aaa4fe9d3b01737b1a15752b1"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/209/files#diff-4b0bf2efa3367c0072ac2bf1e234e703dc46b47aaa4fe9d3b01737b1a15752b1"></a></td><td>&nbsp;</td>
</tr>
</tfoot>
</table>

<sub>

This report was generated by [python-coverage-comment-action](https://github.com/py-cov-action/python-coverage-comment-action)

</sub>
</details>

**Comment by PaulineMauryL** [comment]
Pytest is required instead of unittest, hence it is not as a test in the coverage. I think will should afterwards refactor to get the dashboard code out of the server code. (Ticket 352)

**Comment by PaulineMauryL** [comment]
I might do PR to this PR until it is all good

File: server/lomas_server/administration/tests/test_streamlit_app.py
Comment (LancelotMarti) [inline server/lomas_server/administration/tests/test_streamlit_app.py:1]: nice!

**Review by LancelotMarti — APPROVED** [review]

**Pull request merged**